### DISCUSSION TITLE
android 报错 JNI DETECTED ERROR IN APPLICATION: fid == null

### DISCUSSION THREAD

**Discussion by zengke328** [Tencent/libpag#2379] [Q&A]
【平台】
Android

【版本信息】
 com.tencent.tav:libpag:4.3.51

【问题描述】
最近几个版本Google后台有很多报错信息， JNI DETECTED ERROR IN APPLICATION: fid == null。代码上除了PAG从4.3.47升级到4.3.51外，其他没啥改变，请问下这个报错大概是哪个原因可以引起，我好针对性的去排查。

backtrace:
  #00  pc 0x00000000000799c8  /apex/com.android.runtime/lib64/bionic/libc.so!libc.so (abort+168)
  #01  pc 0x000000000076fba0  /apex/com.android.art/lib64/libart.so!libart.so (art::Runtime::Abort(char const*)+904)
  #02  pc 0x00000000000357d0  /apex/com.android.art/lib64/libbase.so (android::base::SetAborter(std::__1::function<void (char const*)>&&)::$_0::__invoke(char const*)+80)
  #03  pc 0x0000000000034d58  /apex/com.android.art/lib64/libbase.so (android::base::LogMessage::~LogMessage()+352)
  #04  pc 0x00000000003eaa4c  /apex/com.android.art/lib64/libart.so (art::JavaVMExt::JniAbort(char const*, char const*)+3540)
  #05  pc 0x00000000006a2e90  /apex/com.android.art/lib64/libart.so!libart.so (art::JNI<false>::GetLongField(_JNIEnv*, _jobject*, _jfieldID*)+380)
  #06  pc 0x000000000010ec58  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #07  pc 0x00000000000fff54  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #08  pc 0x00000000000ffd38  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #09  pc 0x000000000010f8bc  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #10  pc 0x00000000000ff264  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #11  pc 0x00000000000f8ff0  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #12  pc 0x00000000000f8ad4  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #13  pc 0x00000000000f8704  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #14  pc 0x00000000001d8514  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #15  pc 0x0000000000147930  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #16  pc 0x0000000000147c48  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #17  pc 0x00000000001483a8  /data/app/~~bxYCOZ1MTfQm_SfYdkEcEw==/com.oversea.matemet-XlYyM49DnwBXVexreFszBA==/lib/arm64/libpag.so
  #18  pc 0x00000000000deb44  /apex/com.android.runtime/lib64/bionic/libc.so!libc.so (__pthread_start(void*)+264)
  #19  pc 0x000000000007b2fc  /apex/com.android.runtime/lib64/bionic/libc.so!libc.so (__start_thread+68)

**Comment by kevingpqi123** [comment]
后续返回问题建议将堆栈中的符号解析出来后发出，JNI DETECTED ERROR IN APPLICATION 相关问题建议关注下你们项目中内存问题，比如是否存在 OOM等，内存不足或 FD 泄漏会导致 jni 方法调用失败

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
